gcp_airflow_foundations/base_class/source_template_config.py

```diff
@@ -0,0 +1,168 @@
+from gcp_airflow_foundations.base_class.facebook_table_config import FacebookTableConfig
+from gcp_airflow_foundations.base_class.template_ingestion_options_config import TemplateIngestionOptionsConfig
+from gcp_airflow_foundations.base_class.column_udf_config import ColumnUDFConfig
+
+
+@dataclass
+class SourceTemplateConfig:
+    """
+    Template configuration data class.
+
+    Attributes:
+        table_names : List of table name. Used for Dag Id/s.
+        dest_table_override: Optional target table name. If None, use table_name instead.
+        surrogate_keys : Keys used to identify unique records when merging into ODS.
+        column_mapping : Mapping used to rename columns.
+        cluster_fields: The fields used for clustering. BigQuery supports clustering for both partitioned and non-partitioned tables.
+        landing_zone_table_name_override_template: Optional staging zone table name template.
+        column_casting : Mapping used to cast columns into a specific data type. Note column name uses that of the landing zone table.
+        ods_config : ODS table configuration. See :class:`gcp_airflow_foundations.base_class.ods_table_config.OdsTableConfig`.
+        hds_config : HDS table confidwguration. See :class:`gcp_airflow_foundations.base_class.hds_table_config.HdsTableConfig`.
+        template_ingestion_options: Configuration for template-level ingestion.
+        facebook_table_config: Extra options for ingesting data from the Facebook API.
+        extra_options: Field for storing additional configuration options.
+        start_date : Start date override for DAG
+        start_date_tz : Timezone
+        version : The Dag version for the table. Can be incremented if logic changes.
+        catchup : Passed to a dag [see doc](https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#catchup).
+            Defaults to True. May want to change it to False if Dag version is changed, and we don't want to rerun past dags.
+    """
+    start_date: Optional[str]
+    ingestion_type: IngestionType  # FULL or INCREMENTAL
+    surrogate_keys: Optional[dict]
+    column_mapping: Optional[dict]
+    cluster_fields: Optional[List[str]]
+    column_casting: Optional[dict]
+    new_column_udfs: Optional[dict]
+    hds_config: Optional[HdsTableConfig]
+    facebook_table_config: Optional[FacebookTableConfig]
+    extra_options: dict = field(default_factory=dict)
+    dest_table_override: Optional[str] = "{table}"
+    landing_zone_table_name_override_template: Optional[str] = "{table}"
+    start_date_tz: Optional[str] = "EST"
+    ods_config: Optional[OdsTableConfig] = OdsTableConfig(
+        ods_metadata=OdsTableMetadataConfig(),
+        ods_table_time_partitioning=None,
+        partition_column_name=None,
+    )
+    template_ingestion_options: TemplateIngestionOptionsConfig = TemplateIngestionOptionsConfig(
+        ingest_mode="INGEST_BY_TABLE_NAMES",
```

Comment:
Can we make it an actual enum?
Take a look into ingestion_type.py  as an example
@danielkulikov

Reply:
Updated